types/angular/jqlite.d.ts

```diff
@@ -651,6 +651,7 @@ interface JQuery {
 
     // Undocumented
     length: number;
+    [index: number]: TElement;
```

Comment:
Was actually wondering why this even worked with `@types/jquery` v3 to begin with. It's because `@types/jquery` v3 was using `ArrayLike`'s index signature which is `readonly`. That's no longer the case because #17513 is now merged. I've reran your build to verify this.

I believe changing the index signature to 
```typescript
readonly [index: number]: HTMLElement;
```
should work.

You'll also need to remove the generic type parameter from `JQuery`. I forgot about this on the first review but `@types/angular` only has `TypeScript Version: 2.3` for tests. One of the big reasons `@types/angular` no longer depends on `@types/jquery` is because the introduction of default type parameters broke a lot of builds.